### PR TITLE
[new release] memgraph_kitty and memgraph (1.0)

### PR DESCRIPTION
**Pull request by Gbury**
Display the representation of memory values in the Kitty terminal emulator

- Project page: <a href="https://github.com/gbury/ocaml-memgraph">https://github.com/gbury/ocaml-memgraph</a>
- Documentation: <a href="https://gbury.github.io/ocaml-memgraph/">https://gbury.github.io/ocaml-memgraph/</a>

##### CHANGES:

First release

### CODE REVIEW

**Comment by Gbury** [comment]
So it seems `dune-release` took only the synopsis for the `memgraph_kitty` package as description for this PR.

To complete that, the `memgraph` library allows one to build the graph of the in-memory representation of arbitrary ocaml values and print them in a `.dot` file. `memgraph_kitty` then integrates that with the `kitty` terminal emulator to show these graphs in your toplevel.

File: packages/memgraph/memgraph.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  "Guillaume Bury <guillaume.bury@gmail.com>"
+  "Armaël Guéneau <armael.gueneau@ens-lyon.org>"
+]
+license: "MIT"
+homepage: "https://github.com/gbury/ocaml-memgraph"
+doc: "https://gbury.github.io/ocaml-memgraph/"
+bug-reports: "https://github.com/gbury/ocaml-memgraph/issues"
+depends: [
+  "dune" {>= "2.7"}
+  "ocaml" {>= "4.05.0"}
```
Comment (raphael-proust) [inline packages/memgraph/memgraph.1.0/opam:19]: There seems to be an error when building with some versions of OCaml:

```
#=== ERROR while compiling memgraph.1.0 =======================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/Gbury/ocaml-memgraph/releases/download/v1.0/memgraph-1.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/memgraph.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p memgraph -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/memgraph-9-a915ed.env
# output-file          ~/.opam/log/memgraph-9-a915ed.out
### output ###
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlopt.opt -w -40 -principal -g -I src/core/.memgraph.objs/byte -I src/core/.memgraph.objs/native -intf-suffix .ml -no-alias-deps -open Memgraph -o src/core/.memgraph.objs/native/memgraph__Repr.cmx -c -impl src/core/repr.ml)
# File "src/core/repr.ml", line 248, characters 28-41:
# 248 |       let curried_pointer = Obj.raw_field t offset in
#                                   ^^^^^^^^^^^^^
# Error: Unbound value Obj.raw_field
# Hint: Did you mean set_field?
# (cd _build/default && /home/opam/.opam/4.11/bin/ocamlc.opt -w -40 -principal -g -bin-annot -I src/core/.memgraph.objs/byte -intf-suffix .ml -no-alias-deps -open Memgraph -o src/core/.memgraph.objs/byte/memgraph__Repr.cmo -c -impl src/core/repr.ml)
# File "src/core/repr.ml", line 248, characters 28-41:
# 248 |       let curried_pointer = Obj.raw_field t offset in
#                                   ^^^^^^^^^^^^^
# Error: Unbound value Obj.raw_field
# Hint: Did you mean set_field?
```

My guess is that you need to bump this version constraint:

```suggestion
  "ocaml" {>= "4.12.0"}
```

**Comment by raphael-proust** [comment]
The other errors seem to be internal to the runners so I think it's good to merge.

**Comment by kit-ty-kate** [comment]
Thanks! Could you return the fix to the opam file upstream so that it can be taken into account for next release?

**Comment by Gbury** [comment]
Done in [6b378344d56bc44188cb4fbe03c5a50ba471b866](https://github.com/Gbury/ocaml-memgraph/commit/6b378344d56bc44188cb4fbe03c5a50ba471b866)

**Pull request merged**